code/jobs/CleanupJob.php

```diff
@@ -79,6 +86,13 @@ public function getJobType() {
 	 * Clear out stale jobs based on the cleanup values
 	 */
 	public function process() {
+		// construct limit statement if query_limit is valid int value
+		$limit = '';
+		$query_limit = $this->config()->get('query_limit');
+		if (is_numeric($query_limit) && $query_limit >= 0) {
```

Comment:
This PR looks okay to me, I assume you're hitting an issue where it's trying to clean up too many records at once? The only minor thing that jumps out to me is the `>= 0`. Is 0 a reasonable limit? I guess there may be a use case where someone wants to completely disable this job without actually stopping it from running, so setting it to 0 might make sense.

Reply:
Yea, I mostly just wanted to disallow negative integers and saw no particular reason to disallow 0. Thanks for review, I'll merge it in.